doc/devs.rst

```diff
@@ -0,0 +1,656 @@
+.. _devs:
+
+Developer Guide
+===============
+
+Broker is based on the `C++ Actor Framework (CAF)
+<http://www.actor-framework.org>`_. Experience with CAF certainly helps, but a
+basic understanding of the actor model as well as publish/subscribe suffices for
+understanding (and contributing to) Broker.
+
+In the code base of Broker, we frequently use templates, lambdas, mixins (static
+polymorphism), etc. as well as common C++ idioms such as CRTP and RAII.
+Developers should bring *at least* advanced C++ skills.
+
+Architecture
+------------
+
+From a user perspective, a Broker endpoint is the  primary component in the API
+(see :ref:`endpoint`). Internally, an endoint is a container for an actor system
+that hosts the *core actor* plus any number of *subscribers* and *publishers*.
+The figure below shows a simplified architecture of Broker in terms of actors.
+
+.. figure:: _images/endpoint.png
+  :align: center
+  :alt: Simplified architecture of a Broker endpoint in terms of actors.
+
+A Broker endpoint always contains exactly one core actor. From the perspective
+of the implementation, this actor is the primary component. It manages
+publishers and subscribers, establishes peering relations, forwards messages to
+remote peers, etc.
+
+Because the core actor has many roles to fill, its implementation spreads
+severall classes. The following UML class diagram shows all classes involved in
+implementing the core actor with an exempt of the relevant member functions.
+
+.. figure:: _images/core-actor-uml.png
+  :align: center
+  :alt: All classes involved in implementing the core actor.
+
+
+In a distributed setting, each core actor represents one *peer*.
+
+Multi-Hop Routing / Application-Layer Multicast (ALM)
```

Comment:
I suggest swapping there order of 6.2 ALM and 6.3 Implementation as the Implementation relates pretty closely to the architecture diagram.